### PR TITLE
Add filter based on event criteria for `CountFeaturizer`

### PR DESCRIPTION
**Pull request by ericpan64**
Attempts to resolve https://github.com/som-shahlab/femr/issues/106 -- second pass and rebased onto main!

### CODE REVIEW

**Comment by ericpan64** [comment]
Taking suggestions for what a good test based on the test data might be -- I've been trying to understand the test case and not sure if this code block is affecting the results: https://github.com/som-shahlab/femr/blob/582c5fc75c9ccaeef15448069d1639a165318ae7/src/femr/featurizers/featurizers.py#L322

File: tests/featurizers/test_featurizers.py [hunk 1]
```diff
@@ -126,6 +126,41 @@ def test_count_featurizer(tmp_path: pathlib.Path):
...
+
+    # Test filtering all codes
+    featurizer = CountFeaturizer(excluded_event_filter=lambda _: True, time_bins=[datetime.timedelta(days=0)])
+    featurizer.preprocess(patient, labels)
+    patient_features = featurizer.featurize(patient, labels, ontology)
+
+    assert count_nonempty_columns(patient_features) == 0
+
+    # Test filtering no codes
+    featurizer = CountFeaturizer(excluded_event_filter=lambda _: False, time_bins=[datetime.timedelta(days=0)])
```
Comment (Lalaland) [inline tests/featurizers/test_featurizers.py:155]: lambda _: False seems like a bad way to test this sort of logic. You should probably have a check that passes for just a single event and make sure that one event is dropped.

File: tests/featurizers/test_featurizers.py [hunk 2]
```diff
@@ -126,6 +126,41 @@ def test_count_featurizer(tmp_path: pathlib.Path):
...
+    featurizer.preprocess(patient, labels)
+    patient_features = featurizer.featurize(patient, labels, ontology)
+
+    assert featurizer.get_num_columns() == 3  # Same behavior as previous test
+
+    # TODO: Add test that simulates expected behavior
```
Comment (Lalaland) [inline tests/featurizers/test_featurizers.py:161]: You can check this by looking at patient_features[0], the feature matrix.

**Review by Lalaland — CHANGES_REQUESTED** [review]
I can't seem to leave a comment in the right space, but you are missing a check to the excluded event filter in the second event loop if the featurize method.

> for event in patient.events:
>                code = event.code

File: tests/featurizers/test_featurizers.py (hunk 2 above)
Comment (ericpan64) [reply tests/featurizers/test_featurizers.py:161]: Let's discuss this in-person today -- it's not clear to me how to interpret patient_features

**Comment by ericpan64** [comment]
@Lalaland Updates made to disable time bins for now and added as TODO on board - PR good for re-review!

**Review by Lalaland — APPROVED** [review]

**Comment by Lalaland** [comment]
Feel free to merge when you are ready! Nice work

**Comment by ericpan64** [comment]
Thanks and appreciate the help as well 🙌🙏

**Pull request merged**